### DISCUSSION TITLE
Is there a performance cost to calling TSLexer::get_column?

### DISCUSSION THREAD

**Discussion by ahelwer** [tree-sitter/tree-sitter#1340] [Q&A] [answered]
I recall seeing in an old issue that calling `TSLexer::get_column` has the lexer read backward till it reaches a newline. Is that still the case or does the lexer keep track of the column position as it advances? Similarly, is there any performance impact to calling `TSLexer::mark_end` or any of the other functions?

**Comment by maxbrunsfeld** [comment, marked as answer]
No, `get_column` doesn't do anything expensive anymore. `mark_end` doesn't either.
